pkg/repository/ensurer.go

```diff
@@ -122,8 +122,6 @@ func (r *Ensurer) waitBackupRepository(ctx context.Context, namespace string, ba
 		if err == nil {
 			repo = found
 			return true, nil
-		} else if isBackupRepositoryNotFoundError(err) || isBackupRepositoryNotProvisionedError(err) {
```

Comment:
Once an error is returned from the callback function, would the Poll function return immediately?

Reply:
I'll have to double check. A prior iteration of this kept this part but print error when timeout occurs.

Reply:
Fixed the PR. Err from checkFunc will return on timeout, but not before then.